### PR TITLE
DT-1272 smoke test data

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/SmokeTestHelperResource.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import uk.gov.justice.hmpps.prison.service.SmokeTestHelperService;
+
+import javax.validation.constraints.NotNull;
+
+@RestController
+@Api(tags = {"smoketest"})
+@RequestMapping("${api.base.path}/smoketest")
+@Validated
+@Slf4j
+@ConditionalOnProperty(name = "smoke.test.aware", havingValue = "true")
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/SmokeTestHelperResource.java:26]: Defaults to false everywhere, initially only turned on in t3.

File: src/main/resources/db/migration/data/R__4_11__OFFENDER_IMPRISON_STATUSES.sql [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-        (-4, 1, 'SENT', TO_DATE('2016-03-30', 'YYYY-MM-DD'), TO_DATE('2016-03-30 15:08:00', 'YYYY-MM-DD HH24:MI:SS'), TO_DATE('2019-08-01 11:45:07', 'YYYY-MM-DD HH24:MI:SS'), 'LEI', null,'Y'),
-        (-6, 1, 'SENT', TO_DATE('2016-03-30', 'YYYY-MM-DD'), TO_DATE('2016-03-30 15:08:00', 'YYYY-MM-DD HH24:MI:SS'), TO_DATE('2019-08-01 11:45:07', 'YYYY-MM-DD HH24:MI:SS'), 'LEI', null,'Y'),
-        (-24, 1, 'TRL', TO_DATE('2016-03-30', 'YYYY-MM-DD'), TO_DATE('2016-03-30 15:08:00', 'YYYY-MM-DD HH24:MI:SS'), TO_DATE('2019-08-01 11:45:07', 'YYYY-MM-DD HH24:MI:SS'), 'LEI', null,'Y'),
-        (-27, 1, 'TRL', TO_DATE('2016-03-30', 'YYYY-MM-DD'), TO_DATE('2016-03-30 15:08:00', 'YYYY-MM-DD HH24:MI:SS'), TO_DATE('2019-08-01 11:45:07', 'YYYY-MM-DD HH24:MI:SS'), 'LEI', null,'Y'),
-        (-29, 1, 'TRL', TO_DATE('2016-03-30', 'YYYY-MM-DD'), TO_DATE('2016-03-30 15:08:00', 'YYYY-MM-DD HH24:MI:SS'), TO_DATE('2019-08-01 11:45:07', 'YYYY-MM-DD HH24:MI:SS'), 'LEI', null,'Y'),
+ VALUES (-1, 1, 'SENT', TO_DATE('2017-06-16', 'YYYY-MM-DD'), TO_DATE('2017-06-16 12:27:00', 'YYYY-MM-DD HH24:MI:SS'), null, 'LEI', 'Some Comment Text', 'Y'),
```
Comment (mikehalmamoj) [inline src/main/resources/db/migration/data/R__4_11__OFFENDER_IMPRISON_STATUSES.sql:2]: Correct seed data, confirmed this is correct by checking a real env.

File: src/main/java/uk/gov/justice/hmpps/prison/service/SmokeTestHelperService.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderImprisonmentStatus;
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderImprisonmentStatusRepository;
+import uk.gov.justice.hmpps.prison.security.VerifyBookingAccess;
+
+import java.time.LocalDateTime;
+
+import static java.lang.String.format;
+
+@Service
+public class SmokeTestHelperService {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/SmokeTestHelperService.java:17]: surprised you're writing new classes in java?

File: helm_deploy/values-t3.yaml [hunk 4]
```diff
@@ -14,3 +14,4 @@ env:
   SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_JWK_SET_URI: "https://sign-in-dev.hmpps.service.justice.gov.uk/auth/.well-known/jwks.json"
   SPRING_DATASOURCE_HIKARI_MAXIMUM_POOL_SIZE: 5
   SPRING_REPLICA_DATASOURCE_HIKARI_MAXIMUM_POOL_SIZE: 10
+  SMOKE_TEST_AWARE: true
```
Comment (petergphillips) [inline helm_deploy/values-t3.yaml:17]: needs setting in `_envs.tpl`

**Review by petergphillips — DISMISSED** [review]
LGTM

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**